### PR TITLE
Event sourcing

### PR DESCRIPTION
**Pull request by radresian**

### CODE REVIEW

**Comment by iluwatar** [comment]
@radresian can you resolve the conflict please and I will review.

**Comment by radresian** [comment]
@iluwatar conflict is resolved, also I added some comments on App class, added a Unit Test and replaced System.otuts with log4j logging.

File: event-sourcing/src/main/java/com/iluwatar/event/sourcing/AccountService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+package com.iluwatar.event.sourcing;
+
+import com.iluwatar.event.sourcing.api.EventProcessor;
+import com.iluwatar.event.sourcing.event.AccountCreateEvent;
+import java.util.Date;
+
+/**
+ * Created by Serdar Hamzaogullari on 06.08.2017.
+ */
```
Comment (iluwatar) [inline event-sourcing/src/main/java/com/iluwatar/event/sourcing/AccountService.java]: Can you add proper descriptions for the classes? What are the classes responsibilities. This occurs elsewhere too.

File: event-sourcing/src/main/java/com/iluwatar/event/sourcing/AccountService.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+public class AccountService {
+
+  private EventProcessor eventProcessor;
```
Comment (iluwatar) [inline event-sourcing/src/main/java/com/iluwatar/event/sourcing/AccountService.java]: Fields set only on construction should be final. Check elsewhere too.

File: event-sourcing/src/main/java/com/iluwatar/event/sourcing/MoneyTransactionService.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  /**
+   * Deposit money.
+   *
+   * @param accountNo the account no
+   * @param money the money
+   */
+  public void depositMoney(int accountNo, BigDecimal money) {
+    MoneyDepositEvent moneyDepositEvent = new MoneyDepositEvent(
+        SequenceIdGenerator.nextSequenceId(), new Date().getTime(), accountNo, money);
+    eventProcessor.process(moneyDepositEvent);
```
Comment (iluwatar) [inline event-sourcing/src/main/java/com/iluwatar/event/sourcing/MoneyTransactionService.java]: Seems unnecessary to create the temp variable. Could be written on one line. Check elsewhere for this too.

File: event-sourcing/src/main/java/com/iluwatar/event/sourcing/SequenceIdGenerator.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+   * Next sequence id long.
+   *
+   * @return the long
+   */
+  public static long nextSequenceId() {
+    sequenceId++;
+    return sequenceId;
+  }
+
+}
```
Comment (iluwatar) [inline event-sourcing/src/main/java/com/iluwatar/event/sourcing/SequenceIdGenerator.java]: This class is not meant to be instantiated, right? It should have private constructor.

File: event-sourcing/src/main/test/java/com/iluwatar/event/sourcing/IntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
+/**
```
Comment (iluwatar) [inline event-sourcing/src/main/test/java/com/iluwatar/event/sourcing/IntegrationTest.java]: The path of the test file is funny. It's usually `src/test/java`, right? At the moment `mvn clean install` doesn't run the test at all.

**Comment by iluwatar** [comment]
@radresian review comments given. Please let me know once you've implemented the changes.

**Comment by radresian** [comment]
@iluwatar your change requests are implemented. I removed all optional classes and interfaces to simplify my example.

**Comment by iluwatar** [comment]
Looks perfect @radresian Thank for for the new pattern :+1:

**Pull request merged**